lib/split/user.rb

```diff
@@ -38,28 +48,41 @@ def cleanup_old_versions!(experiment)
 
     def active_experiments
       experiment_pairs = {}
-      user.keys.each do |key|
-        Metric.possible_experiments(key_without_version(key)).each do |experiment|
+      keys_without_finished(user.keys).each do |key|
+        Metric.possible_experiments(Split::Experiment.key_without_version_for_experiment_key(key)).each do |experiment|
           if !experiment.has_winner?
-            experiment_pairs[key_without_version(key)] = user[key]
+            experiment_pairs[Split::Experiment.key_without_version_for_experiment_key(key)] = user[key]
           end
         end
       end
       experiment_pairs
     end
 
+    def key_for_experiment(experiment)
```

Comment:
Should this be named `keys_for_experiment`?  It's returning an array of potentially multiple keys, no?

Reply:
In need to learn how to ruby better :-/

>  find(ifnone = nil) → an_enumerator
> Passes each entry in enum to block. Returns the first for which block is not false.

Reply:
Actually, is "the first key that matches the experiment name" the desired behavior here?  Isn't that just an arbitrary version of the experiment? Wouldn't you want the most recent version?

Reply:
users are only ever put into a version of an experiment at a time. This should be understood as locating the key that a user is holding for an experiment _regardless of the version_

Reply:
So it makes sense that a user should only be in one experiment at a time, but is there somewhere we can point to shows this being enforced?

Reply:
Unfortunately it's enforced implicitly across a few blocks of code. Upon initialization of an experiment, it checks to see if the user has the key, and reads from that key, else writes to it. I wish It were easier to demonstrate as a block of code :\